ethers-providers/src/lib.rs

```diff
@@ -595,12 +595,15 @@ pub trait Middleware: Sync + Send + Debug {
     }
 
     /// Executes the given call and returns a number of possible traces for it
-    async fn debug_trace_call<T: Into<TypedTransaction> + Send + Sync>(
+    async fn debug_trace_call<
+        T: Into<TypedTransaction> + Send + Sync,
+        R: Serialize + DeserializeOwned + Debug + Send,
+    >(
```

Comment:
I'm not sure about this because this would allow deserializing into any type.

what types are possible here?

perhaps we can add some safety trait restrictions here. For example
introduce a trait that is 

```rust
trait CallTrace: DeserializeOwned + Debug + Send {}

impl CallTrace for /*All types we can return in `debug_trace_call`*/ {}

```

Reply:
> I'm not sure about this because this would allow deserializing into any type.

> what types are possible here?

Here are some tricky things with JS tracer. For JS tracer, it could return any type depending on the JS. Look at the example here https://github.com/gakonst/ethers-rs/blob/54879a83484da7a196801ee3a734d7d6c5ed8304/examples/geth_trace_call.rs#L60.

Focus on the `result: function() { return {\"result\":\"HelloWorld\"};` here. For this JS tracer, it would return something like 
```rust
struct Custom{
  result: String
}
```

And you could define some other results like `result: function() { return {\"a\":\"HelloWorld\", \"b\": 123};`  then the return type would be 
```rust
struct Custom{
  a: String,
  b: u64
}
```


> perhaps we can add some safety trait restrictions here.

That's a good idea.

Reply:
I see, so adding another trait bound is likely unnecessary.